app/src/main/java/com/nextcloud/talk/users/UserManager.kt

```diff
@@ -123,6 +123,18 @@ class UserManager internal constructor(private val userRepository: UsersReposito
         }.toSingle()
     }
 
+    fun updateUser(user: User): Single<Int> {
+        return if (user.id != null) {
+            Single.just(userRepository.updateUser(user))
+        } else {
+            Single.just(userRepository.insertUser(user).toInt())
+        }
```

Comment:
```suggestion
        return Single.fromCallable {
            when (user.id) {
                null -> userRepository.insertUser(user).toInt()
                else -> userRepository.updateUser(user)
            }
        }
```
Simplified code a bit.

I'd prefer if we defined a `Single` version of `updateUser/insertUser` and just return a call to that, but this is fine for now.